src/ProxyClient/LiteSpeed.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\RefreshCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+
+/**
+ * LiteSpeed (OpenLiteSpeed or LiteSpeed Web Server) invalidator.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class LiteSpeed extends HttpProxyClient implements PurgeCapable, TagCapable, ClearCapable, RefreshCapable
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        $this->sendPurgeRequest([
+            'X-LiteSpeed-Purge' => '*',
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function purge($url, array $headers = [])
+    {
+        // LiteSpeed only supports purging by relative URLs
+        $urlParts = parse_url($url);
+        $url = array_key_exists('path', $urlParts) ? $urlParts['path'] : '/';
+
+        $this->sendPurgeRequest([
+            'X-LiteSpeed-Purge' => $url,
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function invalidateTags(array $tags)
+    {
+        $this->sendPurgeRequest([
+            'X-LiteSpeed-Purge' => implode(', ', preg_filter('/^/', 'tag=', $tags)),
+        ]);
+
+        return $this;
+    }
+
+    private function sendPurgeRequest(array $headers)
+    {
+        // TODO: LiteSpeed is likely going to hard code this URL, otherwise it has to be configurable
```

Comment:
this says "TODO" - should we merge it like this for now? or can we link to some issue tracker of litespeed?

Reply:
Nooo, this is far from being done! It has to wait until LiteSpeed releases the new features so I can actually integrate it. All we have in this PR is just preparation work :)